Makes sysfs-file 'flags' of mtd writeable

The flags file must be writeable to be able to remove the read-only flag
of mtd blocks before updating them, e.g.

	echo 0x400 > /sys/class/mtd/mtd5/flags
	flashcp -v /tmp/uboot-env.dat /dev/mtd5
	echo 0xc00 > /sys/class/mtd/mtd5/flags




diff --git a/drivers/mtd/mtdcore.c b/drivers/mtd/mtdcore.c
index 73b605577..57bd7f19f 100644
--- a/drivers/mtd/mtdcore.c
+++ b/drivers/mtd/mtdcore.c
@@ -156,7 +156,22 @@ static ssize_t mtd_flags_show(struct device *dev,
 	return snprintf(buf, PAGE_SIZE, "0x%lx\n", (unsigned long)mtd->flags);
 
 }
-static DEVICE_ATTR(flags, S_IRUGO, mtd_flags_show, NULL);
+static ssize_t mtd_flags_store(struct device *dev,
+			       struct device_attribute *attr,
+			       const char *buf, size_t count)
+{
+	struct mtd_info *mtd = dev_get_drvdata(dev);
+	unsigned int flags;
+	int retval;
+
+	retval = kstrtouint(buf, 0, &flags);
+	if (retval)
+		return retval;
+
+	mtd->flags = flags;
+	return count;
+}
+static DEVICE_ATTR(flags, S_IRUGO | S_IWUSR, mtd_flags_show, mtd_flags_store);
 
 static ssize_t mtd_size_show(struct device *dev,
 		struct device_attribute *attr, char *buf)
